src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -156,23 +161,29 @@ public DynamicObject readString(DynamicObject pointer, int length) {
     public static abstract class PointerReadStringToNullPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
         @Specialization(guards = "isNullPointer(pointer)")
-        public DynamicObject readNullPointer(DynamicObject pointer) {
-            return createString(RopeConstants.EMPTY_ASCII_8BIT_ROPE);
+        public DynamicObject readNullPointer(DynamicObject pointer,
+                                             @Cached("create()") AllocateObjectNode allocateObjectNode) {
+            return allocateObjectNode.allocate(coreLibrary().getStringClass(), Layouts.STRING.build(false, false, RopeConstants.EMPTY_ASCII_8BIT_ROPE));
         }
 
         @Specialization(guards = "!isNullPointer(pointer)")
-        public DynamicObject readStringToNull(DynamicObject pointer) {
+        public DynamicObject readStringToNull(DynamicObject pointer,
+                                              @Cached("create()") StringNodes.MakeStringNode makeStringNode) {
+            final byte[] bytes;
+
             if (TruffleOptions.AOT) {
                 final jnr.ffi.Pointer ptr = Layouts.POINTER.getPointer(pointer);
                 final int nullOffset = findNullOffset(ptr);
-                final byte[] bytes = new byte[nullOffset];
+                bytes = new byte[nullOffset];
 
                 PointerOperations.readPointer(ptr, bytes, nullOffset);
 
-                return createString(bytes, ASCIIEncoding.INSTANCE);
+                return makeStringNode.executeMake(bytes, ASCIIEncoding.INSTANCE, CodeRange.CR_UNKNOWN);
```

Comment:
Thanks. That was the intention. I pulled `bytes` up, but forgot to remove the `return`.